### DISCUSSION TITLE
Defining open boundaries

### DISCUSSION THREAD

**Discussion by acroucher** [noaa-ocs-modeling/OCSMesh#58] [Q&A]
The `mesh.boundaries.auto_generate()` function seems to do an OK job of detecting the open ocean boundary (with some nodes possibly omitted at the coastal ends -  if the threshold is increased to pick them up, it starts identifying spurious open boundaries on land, I guess just an inevitable limitation of trying to identify boundaries solely based on elevation).

I will also need to define some open boundaries for e.g. river inputs, and so I probably need to be able to define them a bit more manually.

The user guide states that "To preserve the boundary type accuracy it is possible to programmatically force some
nodes to be on land or in the ocean. However, this is outside the scope of this document." So, how is this done?

**Comment by SorooshMani-NOAA** [comment]
It involves modifying the internal dictionary of boundaries and I think at this point could be really messy. You need to get the `Boundaries` object and modify the `_data` and then call `_init_dataframes` instead of calling `auto_generate` ...
https://github.com/noaa-ocs-modeling/OCSMesh/blob/1fe13b0c0d5040ad84ef3f7ed872b38a7fe329c1/ocsmesh/mesh/mesh.py#L1525-L1551

Frankly, the more I think about it, the worse setting boundaries manually seems to me! Let's start from scratch! 

How would *you* like to set up boundaries manually? 
One way I can think of for doing so is by passing `shapely.geometry..shapes` or `geopandas.GeoDataFrames` and specifying node types based on in/out of shape location.

**Comment by acroucher** [comment]
Yes, I think it would be good to be able to set boundaries based on whether particular nodes are inside a specified shape.

The auto generation of boundaries is good at least as a starting point (detecting internal islands and open sea boundary), so I think I'd just want to be able to modify the ones it detects rather than specify them all manually instead.
